buzsaki_lab_to_nwb/original_code/convert_yuta.m

```diff
@@ -1,162 +1,301 @@
-addpath(genpath(pwd));
 
-generateCore('schema/core/nwb.namespace.yaml');
+addpath(genpath(pwd)); % Cody: when run from the YutaMouse41-150903
+addpath(genpath('D:/GitHub/matnwb'))
+addpath(genpath('D:/GitHub/buzcode')) % Cody: needed for current session read method
+
+%generateCore()
+%generateCore('schema/core/nwb.namespace.yaml');
 
-%% file
 
-fpath = '/Users/bendichter/Desktop/Buzsaki/SenzaiBuzsaki2017/YutaMouse41-150903';
+%% file
 
+%fpath = '/Users/bendichter/Desktop/Buzsaki/SenzaiBuzsaki2017/YutaMouse41-150903';
+fpath = 'D:/BuzsakiData/SenzaiY/YutaMouse41/YutaMouse41-150903';
 
 [fpath_base, fname, ext] = fileparts(fpath);
 session_description = 'mouse in open exploration and theta maze';
 identifier = fname;
-session_start_time = datetime(2015, 7, 31);
+session_start_time = datetime(2015, 7, 31); % (Y,M,D)
 institution = 'NYU';
 lab = 'Buzsaki';
 
-sessionInfo = bz_getSessionInfo(fpath, 'noPrompts', true);
-samplingRate = sessionInfo.lfpSampleRate;
+%sessionInfo = bz_getSessionInfo(fpath, 'noPrompts', true); % Cody: outdated version of retrieving session info
+sessionInfo = LoadParameters('YutaMouse41-150903.xml'); % Cody: this is new version
+samplingRate = sessionInfo.lfpSampleRate; % units? assume kHz
 channel_groups = {sessionInfo.SpkGrps.Channels};
 
 
+% Cody: This is the original code
+%
+%source = fname;
+%file = nwbfile;
+%file.file_create_date = {datestr(now, 'yyyy-mm-dd hh:MM:SS')};
+%file.identifier = {fname};
+%file.session_description = {fname};
+%file.session_start_time = {sessionInfo.Date};
+
+% Cody: This is the new code
+nwb = NwbFile( ...
+    'session_description', 'mouse in open exploration and theta maze', ...
+    'file_create_date', datestr(now, 'yyyy-mm-dd hh:MM:SS'), ...
+    'identifier', fname, ...
+    'session_start_time', session_start_time, ...
+    'general_experimenter', 'Yuta Senzai', ...
+    'general_session_id', sessionInfo.session.name((strfind(sessionInfo.session.name,'-')+1):end), ...
+    'general_institution', 'NYU');
 
-source = fname;
-file = nwbfile;
-file.file_create_date = {datestr(now, 'yyyy-mm-dd hh:MM:SS')};
-file.identifier = {fname};
-file.session_description = {fname};
-file.session_start_time = {sessionInfo.Date};
+
+%%
+nwb.general_subject = types.core.Subject( ...
+    'subject_id', '41', ...
+    'description', 'Yuta mouse 41', ...
+    'genotype', '', ... % unknown
```

Comment:
We should not use empty strings as values. You can remove `'genotype'` entirely because it is optional.

Reply:
Switched to python code - genotype is now identified.